### DISCUSSION TITLE
Why does the `isset()` Operator not require Explicit Types?

### DISCUSSION THREAD

**Discussion by Danaozhong** [ndsev/zserio#617] [Q&A]
The `zserio` specification about bitmasks states the following:
> In expressions outside of the defining type, bitmask values must always be prefixed by the type name and a dot, e.g. Permission.WRITABLE.

This doesn't seem to apply to the `isset()` operator, as the example provided shows that writing the following is possible:
`return isset(testBitmask, INT);`
This example also compiles with `zserio` (tested version 2.11.0).
This code does not compile:
`return ((testBitmask & INT) == TestBitmask.INT);`

The `isset()` example is not within the defining type, but omits the type name and dot. Based on the documentation of bitmask types, I would have expected only the following syntax to be legal:
`return isset(testBitmask, TestBitmask.INT);`

That feels a bit inconsistent to me. Of course, it makes sense that `isset()` can deduce the bitmask type from the first operand, but it seems to [contradict the documentation](https://zserio.org/doc/ZserioLanguageOverview.html#bitmask-types).

**Comment by mikir** [comment]
Thanks pointing this out! I think we forgot to update documentation of `bitmask` types when `isset()` operator had been implemented.

The reason why `isset` operator does not require type name prefix is that it is possible to deduce it from the first operand and because of the brevity. Such deducing is not possible for all expressions. Example:

```valueof(Bitmask.INT)```

But I agree that this is a little bit inconsistent. We might think to improve expressions to allow omitting the name prefix if possible, but I am afraid that this will bring even more confusions. Therefore, I just suggest to update documention for now.
